icu4c/source/test/intltest/units_test.cpp

```diff
@@ -247,9 +247,12 @@ void UnitsTest::testConverter() {
             continue;
         }
 
+        double maxDelta = 1e-6 * uprv_fabs(testCase.expectedValue);
+        if (testCase.expectedValue == 0) {
+            maxDelta = 1e-12;
+        }
         assertEqualsNear(UnicodeString("testConverter: ") + testCase.source + " to " + testCase.target,
```

Comment:
I suggest to make the behaviour of `assertEqualsNear` should be changed to compare the diff percentage with delta not just the diff

Reply:
In this commit: https://github.com/unicode-org/icu/commit/65bbf92f782b1be6da7c4deb56512f9f33a2094e - I changed assertEqualsNear to match the behaviour of JUnit's assertEquals(double, double, double), because the ICU testing functions are modelled on ICU4J. This makes it easier to port testing functions between C++ and Java.

In the rebasing and squashing of commits, originaging in the sffc repository and then the icu-units repository, tracing this back to an original pull request isn't just a few clicks unfortunately.

Here's a pull request in which I originally provide an assertEquals that matches JUnit's behaviour, and we settled on assertEqualsNear (differing from JUnit): https://github.com/sffc/icu/pull/40
(Your review was "I think assertEqualsNear name is more expressive".)

The javadoc for this:
http://junit.sourceforge.net/javadoc/org/junit/Assert.html#assertEquals(java.lang.String,%20double,%20double,%20double)

An absolute delta gives unit tests more control. But I need not argue that it's better, I just need to argue we should be consistent with JUnit and they already made this decision.